### PR TITLE
🌱 Bump CPI to v1.31.0-rc.1

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by chrischdi — COMMENTED** [review]
/lgtm
/approve

File: test/e2e/config/vsphere.yaml [hunk 1]
```diff
@@ -271,7 +271,7 @@ variables:
   KUBERNETES_VERSION_UPGRADE_FROM: "v1.29.0"
   KUBERNETES_VERSION_UPGRADE_TO: "v1.30.0"
   KUBERNETES_VERSION_LATEST_CI: "ci/latest-1.31"
-  CPI_IMAGE_K8S_VERSION: "v1.31.0-rc.0"
+  CPI_IMAGE_K8S_VERSION: "v1.31.0-rc.1"
```
Comment (chrischdi) [inline test/e2e/config/vsphere.yaml:274]: There is no rc.0 image

/approve cancel
/lgtm cancel

Comment (sbueringer) [reply test/e2e/config/vsphere.yaml:274]: Yup waiting for https://github.com/kubernetes/k8s.io/pull/7138

Comment (sbueringer) [reply test/e2e/config/vsphere.yaml:274]: cc: @XudongLiuHarold

Comment (XudongLiuHarold) [reply test/e2e/config/vsphere.yaml:274]: Hi team, the PR just merged, you should be able to pull `v1.31.0-rc.1` now :)

**Comment by sbueringer** [comment]
Tests are green now

**Pull request merged**